update CONTRIBUTING to be on par with the NeIC handbook (#1006)
neicnordic/sensitive-data-archive: CONTRIBUTING.md

# Contributing Guidelines

We thank you in advance 👍 🎉 for taking the time to contribute, whether with *code* or with *ideas*, to the NeIC SDA project.

## Did you find a bug?

> - Ensure that the bug was not already reported by [searching under Issues].

Reviewer: ```suggestion
- Ensure that the bug has not already been reported by [searching under Issues].
```

> - If you're unable to find an (open) issue addressing the problem, open a new one by using the following [template to report a bug]. Be sure to include:

Reviewer: ```suggestion
- If you're unable to find an issue addressing the problem, open a new one by using the following [template to report a bug]. Be sure to include:
```

Reviewer: A bit unclear to use parenthesis (open) here. Better to change to either `an issue` (as I suggested) or `an open issue`.

Author: Better without the `()`.

>       - a *clear* description,
>       - as much relevant information as possible, and
>       - a *code sample* or an (executable) *test case* demonstrating the expected behaviour that is not occurring.

Reviewer 2: ```suggestion
    - a *clear* description,
    - as much relevant information as possible, and
    - a *code sample* or an (executable) *test case* demonstrating the expected behaviour that is not occurring.
```
To fix formatted version.

- If possible, prefix the issue title with a short identifier for the relevant repository component, e.g. **[ingest]**, **[postgres]** etc.

## How to work on a new feature/bug

- Create an issue on Github or you can alternatively pick one already created.

- Assign yourself to that issue.

- Discussions on how to proceed about that issue take place in the comment section on that issue.

> Some of the work might have been done already by a co-worker. In this case, unnecessary work duplication and waste of time and effort are to be avoided. As some of the features might impact different components, it is a good idea to discuss the issue beforehand with the team and communicate the intended changes so that planning can be done accordingly.

Reviewer 2: ```suggestion
To avoid unnecessary work duplication and waste of time and effort, it's generally a good idea to discuss the issue beforehand with team. Some of the work might have been done already by a co-worker. Also, as some of the features might impact different components, please communicate the intended changes, so that planning can be done accordingly.
```

> Some of the work might have been done already by a co-worker. In this case, unnecessary work duplication and waste of time and effort are to be avoided. As some of the features might impact different components, it is a good idea to discuss the issue beforehand with the team and communicate the intended changes so that planning can be done accordingly.

Reviewer 2: ```suggestion
To avoid unnecessary work duplication and waste of time and effort, it's generally a good idea to discuss the issue beforehand with team. Some of the work might have been done already by a co-worker. Also, as some of the features might impact different components, please communicate the intended changes, so that planning can be done accordingly.
```
I found this section a bit hard to follow, so re-ordered it a bit. Feel free to use och discard :)

Author: Reads better this way :-)

## How we work with Git

All work takes place in feature branches. Give your branch a short descriptive name and prefix the name with the most suitable of:

- `feature/`
- `docs/`
- `bugfix/`
- `test/`
- `refactor/`

Use comments in your code, choose variable and function names that clearly show what you intend to implement.

Once the feature is done you can request it to be merged back into `main` by making a Pull Request.

Before making the pull request it is a good idea to rebase your branch to `main` to ensure that eventual conflicts with the `main` branch are solved before the PR is reviewed and that there can be a clean merge.
> NOTE:
> In older github repositories the default branch might be called `master` instead of `main`.

### About git and commit messages

> In general it is better to commit often. Small commits are easier to roll back and also makes the code easier to review.

Reviewer: ```suggestion
In general it is better to commit often. Small commits are easier to roll back and also make the code easier to review.
```

Write helpful commit messages that describe the changes and possibly why they were necessary.

> Each commit should contain changes that are functionally connected and/or related. If, for example, the first line of the commit message contains the word *and*, this is an indicator that this commit should have been split in two.

Reviewer: ```suggestion
Each commit should contain changes that are functionally connected and/or related. If, for example, the first line of the commit message contains the word *and*, this is an indicator that this commit should have been split into two.
```

 6. Wrap the body at 72 characters.
 7. Use the body to explain what and why vs. how.

For an in-depth explanation of the above points, please see [How to Write a Git Commit Message](https://chris.beams.io/posts/git-commit/).

### How we do code reviews

A code review is initiated when someone has made a Pull Request in the appropriate repo on github.

Work should not continue on the branch *unless* it is a [Draft Pull Request](https://github.blog/2019-02-14-introducing-draft-pull-requests/). Once the PR is marked ready the review can start.

> The initiator of the PR should assign the [@sensitive-data-development-collaboration](https://github.com/orgs/neicnordic/teams/sensitive-data-development-collaboration) team for reviewer duty on the branch.

Reviewer 2: ```suggestion
The initiator of the PR should assign the [@sensitive-data-development-collaboration](https://github.com/orgs/neicnordic/teams/sensitive-data-development-collaboration) team as reviews of the branch.
```
